hazelcast/src/main/java/com/hazelcast/sql/SqlColumnType.java

```diff
@@ -67,7 +67,14 @@
     TIMESTAMP_WITH_TIME_ZONE(OffsetDateTime.class),
 
     /** OBJECT type, could be represented by any Java class. */
-    OBJECT(Object.class);
+    OBJECT(Object.class),
+
+    /**
+     * The type of the generic SQL {@code NULL} literal.
+     * <p>
+     * The only valid value of {@code NULL} type is {@code null}.
+     */
+    NULL(Void.class);
```

Comment:
NULL type should be reflected in the design doc.

Reply:
fixed in 80d4cc1b2c7c7f3f9f0349bc8982b83824d9c15a